### PR TITLE
Spec: implement `be_a` and `expect_raises` without macros

### PR DESCRIPTION
**Pull request by asterite**
Code without macros is easier to read.

### CODE REVIEW

File: src/spec/expectations.cr [hunk 1]
```diff
@@ -286,41 +277,40 @@ module Spec
...
-      %failed = false
+    def expect_raises(klass : T.class, message = nil, file = __FILE__, line = __LINE__) forall T
+      failed = false
       begin
-        {{yield}}
-        %failed = true
-        fail "Expected {{klass.id}} but nothing was raised", {{file}}, {{line}}
-      rescue %ex : {{klass.id}}
+        yield
+        failed = true
```
Comment (RX14) [inline src/spec/expectations.cr]: Can't we skip this `failed` hackery and implement it with an `else` clause in the `begin` block? Then we could inline the `begin`/`end` block into the method.

Comment (asterite) [reply src/spec/expectations.cr]: but what about:

```
        fail "Expected #{klass} but nothing was raised", file, line
```

we want to ignore that exception

Comment (asterite) [reply src/spec/expectations.cr]: Thanks, done!

**Comment by Sija** [comment]
This change caused a regression:

```crystal
require "spec"

Int32.should be_a Int32.class
```

v0.24.1:
```
Finished in 5 microseconds
0 examples, 0 failures, 0 errors, 0 pending
```

`master`:
```
Error in line 1: instantiating 'be_a(Class)'

in /.../crystal/src/spec/expectations.cr:271: can't use Object as generic type argument yet, use a more specific type

      Spec::BeAExpectation(T).new
```

**Comment by Sija** [comment]
@asterite @RX14 do you confirm? Or am I trying to do the wrong thing?

**Pull request merged**